lib/api/apiUtils/object/abortMultipartUpload.js

```diff
@@ -66,26 +71,49 @@ function abortMultipartUpload(authInfo, bucketName, objectKey, uploadId, log,
                 // eslint-disable-next-line no-param-reassign
                 request.actionImplicitDenies = originalIdentityAuthzResults;
                 if (err) {
+                    log.error('error aborting MPU', { error: err });
                     return next(err, destBucket);
                 }
                 // for Azure and GCP we do not need to delete data
                 // for all other backends, skipDataDelete will be set to false
-                return next(null, mpuBucket, destBucket, skipDataDelete);
+                return next(null, mpuBucket, destBucket, objectMD, skipDataDelete);
             });
         },
-        function getPartLocations(mpuBucket, destBucket, skipDataDelete,
+        function getPartLocations(mpuBucket, destBucket, objectMD, skipDataDelete,
         next) {
             services.getMPUparts(mpuBucket.getName(), uploadId, log,
                 (err, result) => {
                     if (err) {
+                        log.error('error getting parts', { error: err });
                         return next(err, destBucket);
                     }
                     const storedParts = result.Contents;
-                    return next(null, mpuBucket, storedParts, destBucket,
+                    return next(null, mpuBucket, storedParts, destBucket, objectMD,
                     skipDataDelete);
                 });
         },
-        function deleteData(mpuBucket, storedParts, destBucket,
+        function deleteObjectMetadata(mpuBucket, storedParts, destBucket, objectMD, skipDataDelete, next) {
+            if (!objectMD || metadataValMPUparams.uploadId !== objectMD.uploadId) {
+                return next(null, mpuBucket, storedParts, destBucket, objectMD, skipDataDelete);
+            }
+            // In case there has been an error during cleanup after a complete MPU
+            // (e.g. failure to delete MPU MD in shadow bucket),
+            // we need to ensure that the MPU metadata is deleted.
+            log.info('Object has existing metadata, deleting them', {
```

Comment:
I would put the upload id in the log, if we can have the info, so we can track the problem more easily

Reply:
When we are in this log, both uploadId are the same
https://github.com/scality/cloudserver/pull/5691/files#diff-75b103d02d5ce4b1be1ff37a8a3058d683af6f676c0acef32e4fa1929d23ae7dR96